reVX/least_cost_xmission/least_cost_paths.py

```diff
@@ -0,0 +1,383 @@
+# -*- coding: utf-8 -*-
+"""
+Module to compute least cost xmission paths, distances, and costs one or
+more SC points
+"""
+from concurrent.futures import as_completed
+import geopandas as gpd
+import logging
+import numpy as np
+import os
+import pandas as pd
+from pyproj.crs import CRS
+import rasterio
+import time
+from warnings import warn
+
+from reV.handlers.exclusions import ExclusionLayers
+from rex.utilities.execution import SpawnProcessPool
+from rex.utilities.loggers import log_mem
+
+from reVX.least_cost_xmission.trans_cap_costs import TieLineCosts
+from reVX.utilities.exclusions_converter import ExclusionsConverter
+
+logger = logging.getLogger(__name__)
+
+
+class LeastCostPaths:
+    """
+    Compute least cost paths between desired locations
+    """
+    REQUIRED_LAYRES = ['transmission_barrier']
+
+    def __init__(self, cost_fpath, features_fpath, xmission_config=None):
+        """
+        Parameters
+        ----------
+        cost_fpath : str
+            Path to h5 file with cost rasters and other required layers
+        features_fpath : str
+            Path to geopackage with transmission features
+        resolution : int, optional
+            SC point resolution, by default 128
+        xmission_config : str | dict | XmissionConfig, optional
+            Path to Xmission config .json, dictionary of Xmission config
+            .jsons, or preloaded XmissionConfig objects, by default None
+        """
+        self._check_layers(cost_fpath)
+        self._config = TieLineCosts._parse_config(
```

Comment:
Is calling the private function intentional?

Reply:
Yeah, i probably over coded this. the _parse_config method in TieLineCosts checks for a path or a XmissionConfig object, and if needed creates the XmissionConfig object. We could just require TieLineCosts to take a XmissionConfig object and move the method up to this class, but also it runs instantaneously...